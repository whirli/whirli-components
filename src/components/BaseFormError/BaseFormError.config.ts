// @ts-ignore
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormError/BaseFormError.config';

// Package config
export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  ...config.props,
};
