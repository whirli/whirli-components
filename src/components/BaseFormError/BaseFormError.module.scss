@import '@whirli-components/styles/Breakpoints.module.scss';

// default styles
.error {}
// breakpoint styles
$packageBreakpointStyles: ();