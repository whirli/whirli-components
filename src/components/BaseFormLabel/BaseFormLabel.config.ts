// @ts-ignore
import { PropType } from 'vue';
import { ComponentStyles, ComponentProps } from '../../@types/components';

// Local config
import config from '@whirli-local/components/BaseFormLabel/BaseFormLabel.config';

// Package config
import { PropKeys, PropValues } from './BaseFormLabel.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.NAME]: {
    required: true,
    type: String as PropType<string>,
  },
  [PropKeys.REQUIRED]: {
    required: false,
    type: Boolean as PropType<boolean>,
    default: false,
  },
  ...config.props,
};
