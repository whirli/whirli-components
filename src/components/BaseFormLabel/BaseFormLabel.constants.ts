export const PropKeys: Record<string, string> = {
  REQUIRED: 'required',
  NAME: 'name',
};

export const PropValues: Record<string, string[]> = {};
