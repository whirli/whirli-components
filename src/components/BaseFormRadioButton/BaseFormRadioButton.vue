<template>
  <BaseFormGroupLayoutTertiary :name="id">
    <template #preLabel>
      <slot name="preLabel" />
    </template>
    <template #postInput>
      <slot name="postInput" />
    </template>
    <span :class="wrapperClasses">
      <input
        type="radio"
        :class="inputClasses"
        v-bind="$attrs"
        :disabled="props.disabled"
        :aria-disabled="props.disabled"
        :aria-describedby="getErrorId(props.name, $props.error)"
        :name="props.group"
        :id="id"
        :value="props.value"
        :checked="isChecked"
        @change="updateValue()"
      />
      <span :class="displayClasses" />
      <div ref="renderElement" :aria-checked="props.modelValue" :class="fakerClasses" role="radio">
        <slot name="default" />
      </div>
    </span>
  </BaseFormGroupLayoutTertiary>
</template>

<script setup lang="ts">
// Vue
import { computed, defineEmits, ComputedRef } from '@composition';

// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormRadioButton/BaseFormRadioButton.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormRadioButton.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

// Components
// @ts-ignore
import BaseFormGroupLayoutTertiary from '@whirli-components/components/BaseFormGroup/BaseFormGroupLayoutTertiary.vue';

// Composables
import useClasses from '@whirli-components/@use/class';
import useForm from '@whirli-components/@use/form';

// @ts-ignore
const emit = defineEmits<{
  (event: 'update:modelValue', value: string): void;
  (event: 'reset:errors'): void;
}>();

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
const { makeClasses } = useClasses();
const wrapperClasses: ComputedRef<string[]> = computed(() => [
  styles['radio-button-wrapper'],
  ...makeClasses(ComponentStyles, ConfigProps, props, styles),
]);
const inputClasses: ComputedRef<string[]> = computed(() => [styles['radio-button-input']]);
const displayClasses: ComputedRef<string[]> = computed(() => [styles['radio-button-display']]);
const fakerClasses: ComputedRef<string[]> = computed(() => [styles['radio-button-faker']]);

const isChecked: ComputedRef<boolean> = computed(() => props.value == props.modelValue);
const id: ComputedRef<string> = computed(() => `${props.group}-${props.value}`);

const updateValue = (): void => {
  emit('update:modelValue', props.value);
  emit('reset:errors');
};

const { getErrorId } = useForm();
</script>
