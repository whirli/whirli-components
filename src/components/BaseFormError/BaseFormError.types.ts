import { PropKeys, PropValues } from './BaseFormError.constants';
