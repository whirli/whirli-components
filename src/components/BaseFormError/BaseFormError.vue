<template>
  <BaseText v-bind="$attrs" :class="classes" v-if="$slots.default"><slot /></BaseText>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormError/BaseFormError.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormError.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

// @ts-ignore
const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
import BaseText from '@whirli-components/components/BaseText/BaseText.vue';
const { makeClasses } = useClasses();
const classes = [styles.error, ...makeClasses(ComponentStyles, ConfigProps, props, styles)];
</script>
