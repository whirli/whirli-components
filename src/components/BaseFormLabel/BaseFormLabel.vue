<template>
  <label :for="props.name" :class="[classes, requiredClass]"><slot /></label>
</template>

<script setup lang="ts">
// Styles
// @ts-ignore
import styles from '@whirli-local/components/BaseFormLabel/BaseFormLabel.module.scss?module';

// Data
import { ConfigStyles, ConfigProps } from './BaseFormLabel.config';

// Types
import { ComponentStyles as ComponentStylesInterface } from '@whirli-components/@types/components';
import { Props } from '@whirli-components/@types/props';

const ComponentStyles: ComponentStylesInterface = ConfigStyles;

const props: Props = defineProps(ConfigProps);

// Classes
import useClasses from '@whirli-components/@use/class';
const { makeClasses } = useClasses();
const classes: string[] = [styles.label, ...makeClasses(ComponentStyles, props, styles)];
const requiredClass: string = props.required ? styles.required : '';
</script>
