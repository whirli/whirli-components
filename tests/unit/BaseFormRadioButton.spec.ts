import { mount } from '@vue/test-utils';
import BaseFormRadioButton from '@/components/BaseFormRadioButton/BaseFormRadioButton.vue';
import BaseFormLabel from '@/components/BaseFormLabel/BaseFormLabel.vue';
import BaseFormError from '@/components/BaseFormError/BaseFormError.vue';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
let wrapper: any;
const label = 'Hello World';
const content = 'Hello Content';
const error = 'Hello Error';
const group = 'input-group';
const value = 'input-value';
const id = `${group}-${value}`;

describe('BaseFormRadioButton.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormRadioButton, {
      slots: {
        default: content,
      },
      props: {
        label,
        error,
        group,
        value,
      },
    });
    wrapper.vm.updateValue = jest.fn();
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct label text', () => {
    expect(wrapper.findComponent(BaseFormLabel).text()).toMatch(label);
  });

  it('displays the correct error text', () => {
    expect(wrapper.findComponent(BaseFormError).text()).toMatch(error);
  });

  it('radio has the correct id', () => {
    expect(wrapper.find('input').element.getAttribute('id')).toBe(id);
  });

  it('updates value on change', async () => {
    wrapper.find('input').trigger('change');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.updateValue).toHaveBeenCalled();
  });
});
