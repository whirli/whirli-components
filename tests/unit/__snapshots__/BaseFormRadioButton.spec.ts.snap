// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormRadioButton.vue renders the component 1`] = `
<div
  class=""
>
  
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      <div
        class=""
      >
        
        
        <div
          class=""
        >
          
          <div
            class=""
          >
            
            
          </div>
          <div
            class=""
          >
            
            
            <label
              class=""
              for="input-group-input-value"
            >
              
              Hello World
              
            </label>
            
            
          </div>
          
        </div>
        
        
      </div>
      <div
        class=""
      >
        
        
        <div
          class=""
        >
          
          <div
            class=""
          >
            
            
            
            <span
              class=""
            >
              <input
                class=""
                error="Hello Error"
                id="input-group-input-value"
                label="Hello World"
                name="input-group"
                type="radio"
                value="input-value"
              />
              <span
                class=""
              />
              <div
                class=""
                role="radio"
              >
                
                Hello Content
                
              </div>
            </span>
            
            
            
          </div>
          <div
            class=""
          >
            
            
          </div>
          
        </div>
        
        
      </div>
      
    </div>
    
  </div>
  <div
    class=""
  >
    
    <div
      class=""
    >
      
      
      <span
        class=""
        id="input-group-input-value-error"
      >
        
        
        Hello Error
        
        
      </span>
      
      
    </div>
    
  </div>
  
</div>
`;
